File: HTML/META-INF/resources/dataforms/validator/RequiredValidator.js
/**
 * @fileOverview {@link RequiredValidator}クラスを記述したファイルです。
 */

/**
 * @class RequiredValidator
 * 必須バリデータクラス。
 * <pre>
 * </pre>
 * @extends FieldValidator
 */
RequiredValidator = createSubclass("RequiredValidator", {}, "FieldValidator");

/**
 * バリデーションを行ないます。
 * @param {String} v 値。
 * @returns {Boolean} バリデーション結果。
 */
RequiredValidator.prototype.validate = function(v) {
	return (this.isBlank(v) == false);
};

File: src/dataforms/validator/RequiredValidator.java
package dataforms.validator;


/**
 * 必須バリデータ。
 *
 */
public class RequiredValidator extends FieldValidator {

	/**
	 * コンストラクタ。
	 */
	public RequiredValidator() {
		super("error.required");
	}

	@Override
	public boolean validate(final Object value) {
		return !this.isBlank(value);
	}
}

